src/_pytest/runner.py

```diff
@@ -396,14 +396,75 @@ def pytest_make_collect_report(collector: Collector) -> CollectReport:
 
 
 class SetupState:
-    """Shared state for setting up/tearing down test items or collectors."""
+    """Shared state for setting up/tearing down test items or collectors
+    in a session.
+
+    Suppose we have a collection tree as follows:
+
+    <Session session>
+        <Module mod1>
+            <Function test1>
```

Comment:
Probably better to change the name of `test1` and `test2` to `item1` and `item2`, to better reflect the rest of the text.

Reply:
Oops, fixed.

Thanks for the review!